### PR TITLE
Enable building of Miner Docker for Testnet

### PR DESCRIPTION
**Pull request by lthiery**
* adds BUILD_NET to build pipeline to help in selecting the appropriate genesis block
* creates config for `docker_miner-testnet`

### CODE REVIEW

**Comment by penghjui86** [comment]
Every make is wrong
make: *** [Makefile:33: release] Error 1
root@ubuntu:/home/miner/miner# make release
./rebar3 as prod release -n miner
===> Verifying dependencies...
===> Fetching blockchain (from {git,"https://github.com/helium/blockchain-core.git",
                      {ref,"36ae53877b2aa751ec708b9dc595cb300f1bd1c7"}})
===> Failed to fetch and copy dep: {git,
                                  "https://github.com/helium/blockchain-core.git",
                                  {ref,
                                      "36ae53877b2aa751ec708b9dc595cb300f1bd1c7"}}
make: *** [Makefile:33: release] Error 1

**Comment by lthiery** [comment]
> ./rebar3 as prod release -n miner

This is a different build profile than what you are mentioning. This adds `docker_testminer`

**Comment by lthiery** [comment]
@jadeallenx and/or @jeffgrunewald  
can i get a review on this?

**Review by jeffgrunewald — COMMENTED** [review]
this looks good for the most part but i don't see anything currently triggering the build of a testnet miner that would take advantage of the new release profile added to `rebar.config`; added a question there but it seems to me you'll need to have an idea of a testnet miner tagging strategy similar to how we have `"valX.Y.Z"` for testnet validators and `"validatorX.Y.Z"` for mainnet validators and then add an entry to the `.buildkite/pipeline.yml` that triggers specifically on that tagging pattern and builds the correct release unless you're planning to do that manually or else add it in a later PR.

File: Dockerfile [hunk 1]
```diff
@@ -6,9 +6,9 @@ ARG REBAR_DIAGNOSTIC=0
 ENV DIAGNOSTIC=${REBAR_DIAGNOSTIC}
 
 ARG VERSION
+ARG BUILD_NET
```
[the remark this answers is not in the document]
Comment (lthiery) [reply Dockerfile]: Great idea - thanks!

File: rebar.config [hunk 2]
```diff
@@ -283,6 +283,21 @@
...
                 {include_erts, false}]}]
             },
+            {docker_testminer, [
+              {relx, [
+                {release, {miner, git},
+                  [miner,
+                    tools,
+                    runtime_tools,
+                    recon]},
+                {sys_config, "./config/miner-testnet.config"},
```
[the remark this answers is not in the document]
Comment (lthiery) [reply rebar.config:290]: As you mentioned on Discord: Looks like that’s an OTP-21/rebar3.6 feature that provides “better configuration for releases at runtime” (https://blog.erlware.org/otp-21-new-sys_config_src-option-in-relx/)

I'm doing it this way here just to mirror the setup for mainnet Docker.

File: rebar.config [hunk 3]
```diff
@@ -283,6 +283,21 @@
                 {system_libs, false},
                 {include_erts, false}]}]
             },
+            {docker_testminer, [
+              {relx, [
+                {release, {miner, git},
```
[the remark this answers is not in the document]
Comment (lthiery) [reply rebar.config]: Thanks! Done

**Review by jeffgrunewald — APPROVED** [review]

File: .buildkite/scripts/make_image.sh [hunk 4]
```diff
@@ -53,6 +54,12 @@ case "$BUILD_TYPE" in
         BASE_DOCKER_NAME=$(basename $(pwd))
         DOCKER_NAME="${BASE_DOCKER_NAME}-${IMAGE_ARCH}_${VERSION}"
         ;;
+    "miner-testnet")
+        echo "Doing a testnet miner image build for ${IMAGE_ARCH}"
+        DOCKER_BUILD_ARGS="--build-arg EXTRA_BUILD_APK_PACKAGES=apk-tools --build-arg EXTRA_RUNNER_APK_PACKAGES=apk-tools --build-arg BUILDER_IMAGE=${BASE_IMAGE} --build-arg RUNNER_IMAGE=${BASE_IMAGE} --build-arg REBAR_BUILD_TARGET=docker_testminer ${DOCKER_BUILD_ARGS}"
```
Comment (lthiery) [inline .buildkite/scripts/make_image.sh:59]:
```suggestion
        DOCKER_BUILD_ARGS="--build-arg EXTRA_BUILD_APK_PACKAGES=apk-tools --build-arg EXTRA_RUNNER_APK_PACKAGES=apk-tools --build-arg BUILDER_IMAGE=${BUILD_IMAGE} --build-arg RUNNER_IMAGE=${RUN_IMAGE} --build-arg REBAR_BUILD_TARGET=docker_testminer ${DOCKER_BUILD_ARGS}"
```

File: .buildkite/scripts/make_image.sh [hunk 5]
```diff
@@ -53,6 +54,12 @@ case "$BUILD_TYPE" in
         BASE_DOCKER_NAME=$(basename $(pwd))
         DOCKER_NAME="${BASE_DOCKER_NAME}-${IMAGE_ARCH}_${VERSION}"
         ;;
+    "miner-testnet")
+        echo "Doing a testnet miner image build for ${IMAGE_ARCH}"
+        DOCKER_BUILD_ARGS="--build-arg EXTRA_BUILD_APK_PACKAGES=apk-tools --build-arg EXTRA_RUNNER_APK_PACKAGES=apk-tools --build-arg BUILDER_IMAGE=${BUILD_IMAGE} --build-arg RUNNER_IMAGE=${RUN_IMAGE} --build-arg REBAR_BUILD_TARGET=docker_testminer ${DOCKER_BUILD_ARGS}"
+        BASE_DOCKER_NAME=$(basename $(pwd))
+        DOCKER_NAME="${BASE_DOCKER_NAME}-${IMAGE_ARCH}_${VERSION}"
```
Comment (jeffgrunewald) [inline .buildkite/scripts/make_image.sh:61]:
```suggestion
        DOCKER_NAME="${BASE_DOCKER_NAME}-${IMAGE_ARCH}_testnet_${VERSION}"
```
see the difference between the mainnet and testnet validator builds above between lines 42 and 49; this line is currently identical to the mainnet miner image `DOCKER_NAME` on line 55 so these would likely overwrite each other on quay when they get pushed at the bottom of the file.

**Review by jeffgrunewald — APPROVED** [review]

**Pull request merged**